be/src/service/service_be/starrocks_be.cpp

```diff
@@ -183,6 +183,15 @@ void start_be(const std::vector<StorePath>& paths, bool as_cn) {
     if (config::brpc_num_threads != -1) {
         options.num_threads = config::brpc_num_threads;
     }
+    const auto lake_service_max_concurrency = config::lake_service_max_concurrency;
+    const auto service_name = "starrocks.lake.LakeService";
+    const auto methods = {"abort_txn",           "abort_compaction", "compact",          "drop_table",
+                          "delete_data",         "delete_tablet",    "get_tablet_stats", "publish_version",
+                          "publish_log_version", "vacuum",           "vacuum_full"};
+    for (auto method : methods) {
+        brpc_server->MaxConcurrencyOf(service_name, method) = lake_service_max_concurrency;
```

Comment:
does the brpc have EAGAIN or BACKOFF error to indicate its service overload, this max concurrency is hard to use or properly use in practice.

Reply:
Yes https://github.com/apache/brpc/blob/master/docs/en/error_code.md#error-code-in-brpc

Reply:
in such case, an api timeout with fast fail on server side might be a better choice?

Reply:
> in such case, an api timeout with fast fail on server side might be a better choice?

This may be acceptable for tasks with relatively short timeouts, but for those with very long timeouts like compaction (default 1 day), it may not be that meaningful. Additionally, if a deadlock exists in the server program, it may not even be possible to check for timeout.  Perhaps it would be better to use both mechanisms together.

Reply:
> for those with very long timeouts like compaction (default 1 day), it may not be that meaningful.

It may not good to design a sync RPC call for such use case. Might be better break it into async call by submitting a task and then continue querying the task execution status.

Reply:
> > for those with very long timeouts like compaction (default 1 day), it may not be that meaningful.
> 
> It may not good to design a sync RPC call for such use case. Might be better break it into async call by submitting a task and then continue querying the task execution status.

Yes, it's a worthy improvement. But even if we make this modification, the task may not be able to end for a long time or be canceled due to deadlocks, IO blocking, etc., and may still result in a lot of RPC tasks blocking in the queue, if max_concurrency is not set.